src/corehost/common/pal.h

```diff
@@ -173,6 +174,8 @@ namespace pal
     inline void file_vprintf(FILE* f, const char_t* format, va_list vl) { ::vfprintf(f, format, vl); ::fputc('\n', f); }
     inline void err_vprintf(const char_t* format, va_list vl) { ::vfprintf(stderr, format, vl); ::fputc('\n', stderr); }
     inline void out_vprintf(const char_t* format, va_list vl) { ::vfprintf(stdout, format, vl); ::fputc('\n', stdout); }
+    inline void str_vprintf(char_t* str, size_t size, const char_t* format, va_list vl) { ::vsnprintf(str, size, format, vl); }
```

Comment:
These should return `int` to allow calculating the required buffer size
```C++
inline int str_vprintf(char_t* str, size_t size, const char_t* format, va_list vl) 
{ 
    return ::vsnprintf(str, size, format, vl); 
}
```

Reply:
unfortunately I wasn't able to find the `WCHAR` equivalent of `vsnprintf`. What should be the equivalent `_vsnwprintf` doesn't return the number of characters needed if the buffer is too small, it just returns -1.

So I had to introduce two functions on windows `_vscwprintf` which is just "get me the count" and then `_vsnwprintf` which is the "print it out". On Linux these both are `vsnprintf`.

Reply:
Turns out that the Windows version works as the unix one, it's just not very well documented (the documentation is confusing to say the least). So I simplified it now.